src/pages-and-resources/discussions/app-config-form/apps/shared/discussion-topics/DiscussionTopics.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { Add } from '@edx/paragon/icons';
+import { Button } from '@edx/paragon';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { FieldArray, useFormikContext } from 'formik';
+import { v4 as uuid } from 'uuid';
+
+import messages from '../messages';
+import TopicItem from './TopicItem';
+import { removeModel } from '../../../../../../generic/model-store';
+import { updateDiscussionTopicIds } from '../../../../data/slice';
+import { updatedDiscussionTopics } from '../../../../data/thunks';
+
+const DiscussionTopics = ({ intl }) => {
+  const dispatch = useDispatch();
+  const { values } = useFormikContext();
+  const [topics, setTopics] = useState(values.discussionTopics);
+
+  useEffect(() => {
+    const updatedDiscussionTopicIds = values.discussionTopics?.map(topic => topic.id);
+
+    setTopics(values.discussionTopics);
+    dispatch(updateDiscussionTopicIds(updatedDiscussionTopicIds));
+    dispatch(updatedDiscussionTopics(values.discussionTopics));
+  }, [values.discussionTopics]);
+
+  const handleTopicDelete = (topicIndex, topicId, remove) => {
+    remove(topicIndex);
+    dispatch(removeModel({ modelType: 'discussionTopics', id: topicId }));
+  };
+
+  const addNewtopic = (push) => {
+    const payload = { name: '', id: uuid() };
+    push(payload);
+  };
+
+  return (
+    <>
+      <h5 className="text-gray-500 mt-4 mb-2">
+        {intl.formatMessage(messages.discussionTopics)}
+      </h5>
+      <label className="text-primary-500 mb-2 h4">
+        {intl.formatMessage(messages.discussionTopicsLabel)}
+      </label>
+      <div className="small mb-4 mt-0 text-muted">
```

Comment:
Why mt-0? In general it is a good practise not to use margin-topic. use margin-bottom instead. This way every component manages the margin below itself and the margin top is handled by element above. It keeps the margins consistent and easy to manage.

Reply:
Good point. I removed it and using mb for consistency.